src/hooks/useLoadData.ts

```diff
@@ -0,0 +1,41 @@
+import { Dispatch, useEffect } from "react";
+import { Action } from "../appState";
+
+export const useLoadData = (dispatch: Dispatch<Action>): void => {
+  useEffect(() => {
+    const fetchData = async () => {
+      try {
+        const response = await fetch("/fruits.txt");
+        const text = await response.text();
+
+        const wordPack = text
+          .split("\n")
+          .map((word) => word.trim())
+          .filter(Boolean);
+
+        setTimeout(() => {
+          dispatch({ type: "load-data", wordPack });
+        }, 3000);
```

Comment:
The timeout can most likely be omitted now that we've tested the loading state 😄 

```suggestion
        dispatch({ type: "load-data", wordPack });
```